src/main/java/dev/joopie/jambot/api/youtube/ApiYouTubeService.java

```diff
@@ -9,75 +10,159 @@
+    }
+
+    private List<SearchResultDto> getFilteredVideos(List<String> videoIds, Duration minDuration, Duration maxDuration, List<String> artistNames) throws IOException {
+        String url = String.format(VIDEO_DETAILS_URL, String.join(",", videoIds), properties.getToken());
+        try (var client = HttpClients.createDefault();
+             var response = client.execute(RequestBuilder.get(url).build())) {
+
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                List<SearchResponse.Item> items = getVideoDetailsFromResponse(response.getEntity());
+
+                List<SearchResultDto> matchingChannels = items.stream()
+                        .filter(item -> artistNames.stream().anyMatch(artist -> item.getSnippet().getChannelTitle().toLowerCase().contains(artist.toLowerCase())))
+                        .map(this::mapItemToSearchResult)
+                        .toList();
+
+                if (!matchingChannels.isEmpty()) {
+                    return matchingChannels;
+                }
+
+                return items.stream()
+                        .filter(item -> {
+                            Duration videoDuration = parseDuration(item.getContentDetails().getDuration());
+                            return videoDuration.compareTo(minDuration) >= 0 && videoDuration.compareTo(maxDuration) <= 0;
+                        })
+                        .sorted((item1, item2) -> {
+                            Duration duration1 = parseDuration(item1.getContentDetails().getDuration());
+                            Duration duration2 = parseDuration(item2.getContentDetails().getDuration());
+                            return Long.compare(Math.abs(duration1.minus(minDuration).getSeconds()), Math.abs(duration2.minus(minDuration).getSeconds()));
+                        })
+                        .map(this::mapItemToSearchResult)
+                        .toList();
+            } else {
+                log.warn("Invalid HTTP response status: {}", response.getStatusLine().getStatusCode());
+            }
+        }
+
+        return List.of();
+    }
+
+    private List<SearchResponse.Item> getVideoDetailsFromResponse(HttpEntity entity) throws IOException {
+        SearchResponse response = parseResponse(entity, SearchResponse.class);
+        return Optional.ofNullable(response)
+                .map(SearchResponse::getItems)
+                .orElse(List.of());
     }
 
-    private SearchResponse mapHttpEntityToSearchResponse(final HttpEntity httpEntity) {
-        try {
-            return objectMapper.readValue(httpEntity.getContent(), SearchResponse.class);
-        } catch (IOException exception) {
-            throw new JambotYouTubeException("Whoeps, we couldn't handle that thick search response.", exception);
+    private <T> T parseResponse(HttpEntity entity, Class<T> valueType) throws IOException {
+        try (InputStream content = getDecompressedInputStream(entity);
```

Comment:
Wat is precies de reden dat je dit hebt omgebouwd?

Oh I guess gzip shizzle..

Edit: Heeft de Http client geen GZIP support?

Reply:
HttpClient has support for this. Implemented it now :)